pkg/parser/terraform/terraform.go

```diff
@@ -47,7 +48,32 @@ func (p *Parser) Parse(path string, content []byte) ([]model.Document, error) {
 
 	fc, parseErr := p.convertFunc(file, inputVariableMap)
 
-	return []model.Document{fc}, errors.Wrap(parseErr, "failed terraform parse")
+	json := []model.Document{fc}
+
+	var resourcesElements model.Document
+	var elements model.Document
+	var certInfo map[string]interface{}
+
+	for _, documents := range json { // iterate over documents
+		if documents["resource"] != nil {
+			for _, resources := range documents["resource"].(model.Document) { // iterate over resources
+				resourcesElements = resources.(model.Document)
+				for _, v2 := range resourcesElements { // resource name
+					elements = v2.(model.Document)
+					for k, v3 := range elements { // resource elements
+						if k == "certificate_body" {
+							certInfo = additional.AddCertificateInfo(path, v3.(string))
+							if certInfo != nil {
+								elements["certificate_body"] = certInfo
+							}
+						}
+					}
+				}
+			}
+		}
+	}
```

Comment:
This entire block could be a performance bottleneck. We need to investigate resource consumption and compare it to a baseline.

Take a look at @joaoReigota1 's PR for resource consumption metrics #2961

See if you can merge his code to your branch, or wait for this branch to be merged (probably next week) so you can rebase your changes on top of the updated master. Then you need to provide evidence that this feature will not take a critical hit into the engine's performance.